docs/enhancements/0-single-daemonset.md

```diff
@@ -0,0 +1,278 @@
+- pull secret, if specified;
+- module loading order, if specified;
+- list of firmware files, if specified.
+
+When instructed to by the control-plane, daemon pods keep trying to load or unload kernel modules.
+The control-plane only adds or removes the `kmm.node.kubernetes.io/$namespace.$module.ready` label when it receives the
+appropriate success event from the daemon pods, which guarantees a consistent label.
+
+Locally, daemon Pod should maintain a state with the same lifetime as the kernel modules it loads.
+It is desirable for this state to persist during application restarts, but not node reboots, because by definition all
+KMM-loaded kmods are gone as well.  
+We propose to write that state in a temporary filesystem such as `/run`, which:
+- is not garbage-collected at runtime like `/tmp` is;
+- is emptied during reboots.
+
+#### Connectivity
+
+The control-plane and the daemon Pods communicate over the network, using [gRPC] over TLS.  
+Although connections are initiated by the daemon Pods, communication is bi-directional:
+- daemon Pods send request their config from and send events to the control-plane;
+- the control-plane notifies daemon Pods by sending updates.
+
+This can be achieved with either a combination of
+[Unary RPCs](https://grpc.io/docs/what-is-grpc/core-concepts/#unary-rpc) and
+[Server streaming RPC](https://grpc.io/docs/what-is-grpc/core-concepts/#server-streaming-rpc), or
+[Bidirectional streaming RPCs](https://grpc.io/docs/what-is-grpc/core-concepts/#bidirectional-streaming-rpc).
+
+#### Security
+
+##### Preventing external connections to the control-plane
+
+The control-plane runs sensitive tasks, such as node labeling, and distributes secret data, such as pull secrets, to the
+daemon pods.  
+To make sure that only daemon Pods can communicate with the control-plane, we leverage
+[cert-manager's CSI Driver](https://cert-manager.io/docs/usage/csi/) to authenticate both components.  
+Further restrictions may be configured at the TCP level using a
+[Network Policy](https://kubernetes.io/docs/concepts/services-networking/network-policies/); unfortunately not all
+[network plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) support
+them.
+
+##### Restricting daemon Pods' privileges
+
+Daemon Pods run on all nodes.
+If one of those nodes is compromised, then an attacker could use the Pod's credentials against the API Server for
+malicious purposes.
+We intend to run the daemon Pods with a dedicated ServiceAccount with zero API Server privileges.
+The only sensitive data that the daemon Pod will potentially be processing is pull secrets from other namespaces, which
+will only be used to pull the kmod image once and will not be persisted in any way.
+
+### Hub & Spoke: one proxy in each Spoke
+
+In Hub & Spoke setups, we introduce a third KMM component, the proxy, which is both a [gRPC] proxy for the daemon pods
```

Comment:
So we stop using `ClusterClaims` and start using gRPC instead for everything needed?